futures-util/src/future/mod.rs

```diff
@@ -191,40 +192,35 @@ pub trait FutureExt: Future {
     /// wrapped version of them.
     ///
     /// Also note that if both this and the second future have the same
-    /// success/error type you can use the `Either::split` method to
+    /// output type you can use the `Either::factor_first` method to
     /// conveniently extract out the value at the end.
     ///
     /// # Examples
     ///
     /// ```
-    /// use futures::future::{self, Either};
+    /// #![feature(futures_api)]
+    /// use futures::future::{Either, Future, FutureExt};
     ///
     /// // A poor-man's join implemented on top of select
     ///
-    /// fn join<A, B, E>(a: A, b: B) -> Box<Future<Item=(A::Item, B::Item), Error=E>>
-    ///     where A: Future<Error = E> + 'static,
-    ///           B: Future<Error = E> + 'static,
-    ///           E: 'static,
+    /// fn join<A, B, E>(a: A, b: B) -> impl Future<Output=(A::Output, B::Output)>
+    ///     where A: Future + Unpin,
+    ///           B: Future + Unpin,
     /// {
-    ///     Box::new(a.select(b).then(|res| -> Box<Future<Item=_, Error=_>> {
-    ///         match res {
-    ///             Ok(Either::Left((x, b))) => Box::new(b.map(move |y| (x, y))),
-    ///             Ok(Either::Right((y, a))) => Box::new(a.map(move |x| (x, y))),
-    ///             Err(Either::Left((e, _))) => Box::new(future::err(e)),
-    ///             Err(Either::Right((e, _))) => Box::new(future::err(e)),
+    ///     a.select(b).then(|either| {
+    ///         match either {
+    ///             Either::Left((x, b)) => b.map(move |y| (x, y)).left_future(),
+    ///             Either::Right((y, a)) => a.map(move |x| (x, y)).right_future(),
     ///         }
-    ///     }))
+    ///     })
     /// }
     /// ```
-    fn select<B>(self, other: B) -> Select<Self, B::Future>
-        where B: IntoFuture, Self: Sized
+    fn select<B>(self, other: B) -> Select<Self, B>
```

Comment:
We just changed the `.join` methods into free functions-- seems like that would make sense for `select` as well?

Reply:
Thanks for pointing that out, it makes sense. And I remembered that I forgot to write PR for changing `StreamExt::select` to a function 😅 .